camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/TagsYamlTest.java

```diff
@@ -0,0 +1,171 @@
+                "services:",
+                "- type: " + BasicApplication.class.getName(),
+                "  brooklyn.tags:",
+                "  - tag1",
+                "  - 3");
+        assertTrue(getTagsEventually(app).contains(BrooklynTags.newCampBrooklynTag(3)));
+        assertTrue(getTagsEventually(app).contains(BrooklynTags.newCampBrooklynTag("tag1")));
+    }
+
+    @Test
+    public void testBrooklynCampObjectTags() throws Exception {
+        final Entity app = createAndStartApplication(
+                "services:",
+                "- type: " + BasicApplication.class.getName(),
+                "  brooklyn.tags:",
+                "  - tag1",
+                "  - $brooklyn:object:",
+                "      type: " + TagsTestSerializableObject.class.getName());
+        assertTrue(Iterables.any(getTagsEventually(app), new Predicate<Object>() {
+            @Override
+            public boolean apply(@Nullable Object input) {
+                return input instanceof BrooklynTags.CampBrooklynTag && ((BrooklynTags.CampBrooklynTag)input).getContents() instanceof TagsTestSerializableObject;
+            }
+        }));
+        assertTrue(getTagsEventually(app).contains(BrooklynTags.newCampBrooklynTag("tag1")));
+    }
+
+    @Test
+    public void testBrooklynCampFailDslTags() throws Exception {
+        try {
+            final Entity app = createAndStartApplication(
+                    "services:",
+                    "- type: " + BasicApplication.class.getName(),
+                    "  brooklyn.tags:",
+                    "  - tag1",
+                    "  - $brooklyn:object:",
+                    "      type: "+TagsTestSerializableObject.class.getName(),
+                    "      constructor.args:",
+                    "      - $brooklyn:attributeWhenReady(\"host.name\")");
+            fail("Should throw IllegalArgumentException exception.");
+        } catch (CompoundRuntimeException e) {
+            Asserts.assertStringContainsAtLeastOne(Exceptions.getFirstInteresting(e).getMessage(),"brooklyn.tags should not contain DeferredSupplier objects, such example is using $brooklyn:attributeWhenReady");
+        }
+    }
+
+    @Test
+    public void testBrooklynCampFailNonSerializableDslTags() throws Exception {
+        try {
+            final Entity app = createAndStartApplication(
+                    "services:",
+                    "- type: " + BasicApplication.class.getName(),
+                    "  brooklyn.tags:",
+                    "  - tag1",
+                    "  - $brooklyn:object:",
+                    "      type: "+NonSerializable.class.getName());
+            fail("Should throw IllegalArgumentException exception.");
+        } catch (CompoundRuntimeException e) {
+            Asserts.assertStringContainsAtLeastOne(Exceptions.getFirstInteresting(e).getMessage(),"brooklyn.tags should not contain DeferredSupplier objects, such example is using $brooklyn:attributeWhenReady");
```

Comment:
This is a strange error message to assert - the test has passed in a non-serializable object, rather than a `DeferredSupplier`.